Bottom: 7c9a76f8b5d8d63e3c5ca7b2c5a3c7014a6dd899
Top:    6ca3a5131e6a09a5a225906d8bf2307ebd31c174
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2012-04-10 22:10:26 -0400

Refresh of device-trees.patch

---

diff --git a/drivers/spi/spi-lpc313x.c b/drivers/spi/spi-lpc313x.c
index 0157806..1c45d8f 100644
--- a/drivers/spi/spi-lpc313x.c
+++ b/drivers/spi/spi-lpc313x.c
@@ -48,8 +48,22 @@
 #include <mach/gpio.h>
 
 /* Register access macros */
-#define spi_readl(reg) __raw_readl(&SPI_##reg)
-#define spi_writel(reg,value) __raw_writel((value),&SPI_##reg)
+#define spi_readl(reg) _spi_readl(&SPI_##reg)
+#define spi_writel(reg,value) _spi_writel((value),&SPI_##reg)
+
+static inline void _spi_writel(uint16_t value, volatile void *reg)
+{
+	printk("spi_write %p value %x\n", reg, value);
+	__raw_writew(value, reg);
+}
+
+static inline uint16_t _spi_readl(volatile void *reg)
+{
+	uint16_t value;
+	value = __raw_readw(reg);
+	printk("spi_read %p value %x\n", reg, value);
+	return value;
+}
 
 struct lpc313xspi
 {
